dev_guide/builds.adoc

```diff
@@ -191,6 +191,77 @@ See the xref:../creating_images/s2i.adoc#creating-images-s2i[S2I Requirements] t
 on how to create a builder image supporting incremental builds.
 ====
 
+
+[[extended_builds]]
+
+=== Extended builds
+
+For compiled languages (Go, C, C++, Java, etc.) the dependencies necessary for
+compilation might dramatically increase the size of the image or introduce
+vulnerabilities that can be exploited.
+
+To avoid these problems, source-to-image introduces a dual image build process
+that allows an application to be built as usual - in the builder image, and
+then inject - in the runtime image, the resulted binary artifacts into the
+desired location to be executed.
+
+In order to offer even more flexibility in this process, S2I will execute an
+`assemble-runtime` script inside the runtime image that will allow you to
+customize further the runtime image.
+
+Official documentation for source-to-image extended builds can be found
+https://github.com/openshift/source-to-image/blob/master/docs/runtime_image.md[here].
+
+====
+[source,yaml]
+----
+strategy:
+  type: "Source"
+  sourceStrategy:
+    from:
+      kind: "ImageStreamTag"
+      name: "resulted-image:latest"
+    runtimeImage: <1>
+      kind: "ImageStreamTag"
+      name: "runtime-image:latest"
+    runtimeArtifacts: <2>
+      sourcePath: "/path/to/source"
```

Comment:
Looks invalid because `runtimeArtifacts` contains a list of mappings and I don't see a hyphen here.